clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -162,15 +171,20 @@ void runOnce() {
     private void processApplicationEvents() {
         LinkedList<ApplicationEvent> events = new LinkedList<>();
         applicationEventQueue.drainTo(events);
+        kafkaConsumerMetrics.ifPresent(metrics -> metrics.recordApplicationEventQueueSize(applicationEventQueue.size()));
```

Comment:
Once we drain `applicationEventQueue`, it'll be empty, right? If, so why not just do:

```suggestion
        kafkaConsumerMetrics.ifPresent(metrics -> metrics.recordApplicationEventQueueSize(0));
```